gossip/src/crds_gossip_pull.rs

```diff
@@ -482,6 +482,11 @@ impl CrdsGossipPull {
             let out: Vec<_> = crds
                 .filter_bitmask(filter.mask, filter.mask_bits)
                 .filter(pred)
+                .filter(|entry| {
+                    // Exclude the new ContactInfo until the cluster has
+                    // upgraded.
+                    !matches!(&entry.value.data, CrdsData::ContactInfo(_))
```

Comment:
I have included an "Update Plan" in the PR description.
This patch makes it possible for nodes to ingest any incoming new `ContactInfo` but does not send any new `ContactInfo` over the cluster.
Once all nodes have updated to this patch, we can remove this code and start sending new `ContactInfo` over the cluster. It shouldn't cause any synchronization issue at that point since the receiving nodes would be able to deserialize and process the type correctly.